lib/main.dart

```diff
@@ -54,17 +53,32 @@ class FishPage extends StatefulWidget {
 class FishPageState extends State<FishPage> {
   bool _audioToolsReady = false;
   DocumentSnapshot _lastFish;
+  List<DocumentSnapshot> availableFish;
+  List<DocumentSnapshot> reservedFish;
+  ListType listType;
 
-  FishPageState();
+  FishPageState() {
+    listType = ListType.available;
+  }
 
   @override
   void initState() {
     super.initState();
     if (!_audioToolsReady) populateAudioTools();
+    _createStream(ListType.available).listen((data) {
+      setState(() {
```

Comment:
super nit: consider making this a one-liner:
setState(() => availableFish = data);
(and below)

Reply:
Done!